lib/chef/application/base.rb

```diff
@@ -363,7 +363,7 @@ def fetch_recipe_tarball(url, path)
     Chef::Log.trace("Download recipes tarball from #{url} to #{path}")
     if File.exist?(url)
       FileUtils.cp(url, path)
-    elsif url =~ URI.regexp
+    elsif url&.match?(URI.regexp)
```

Comment:
This nil check could be avoided by flipping these around.

```suggestion
    elsif URI.regexp.match?(url)
```

Reply:
Fixed